shard/init.lua

```diff
@@ -114,6 +113,57 @@ local function wait_connection()
     end
 end
 
+--[[
+Correct way to handle connection fiber
+Arguments:
+    1) url in host:port format
+    2) optional arguments (as Lua table)
+        - user
+        - password
+        - timeout in seconds
+        - reconnect_after in seconds (default: 1)
+    If timeout is nil, fiber will not wait for the connection
+Usage:
+    net_connect('localhost:3301', { user = 'admin', password = '123', timeout = 10 })
+    net_connect('localhost:3301') -- do not wait guest login
+Return:
+    1) ok - true/false
+    2) net.box connection object (ok == true) OR an error message (ok == false)
+]]--
+local function net_connect(uri, opts)
+    local ok, conn = pcall(remote.new, remote, uri, {
+            wait_connected = false,
+            reconnect_after = opts.reconnect_after or 1,
+            user = opts.user,
+            password = opts.password
+    })
+    if not ok then
+        return false, conn
+    end
+    local tm = opts.timeout or 0
+    if not conn:wait_connected(tm) then
+        local _error = conn.error
+        conn:close()
+        return false, _error
+    end
+    return true, conn
+end
+
+local function make_error(errno, fmt, ...)
+   if type(fmt) ~= 'string' then
+        fmt = tostring(fmt)
+    end
+
+    if select('#', ...) ~= 0 then
+        local stat
+        stat, fmt = pcall(string.format, fmt, ...)
+        if not stat then
+            error(fmt, 2)
+        end
+    end
+
+    return nil, { errno = errno, error = fmt }
```

Comment:
I think what this function should return the error object only.

Reply:
I believe this is done to support our error handling procedure, which is:

`return nil, error`
or
`return obj`